src/myprofile-page/MyProfilePage.js

```diff
@@ -0,0 +1,194 @@
+import React, { useContext, useEffect, useState } from 'react';
+import { useMutation, useQuery } from '@apollo/client';
+import { Button } from '@material-ui/core';
+import { AuthContext } from '../context/AuthProvider';
+import {
+  GET_COUNTRIES, GET_USER, UPDATE_CONTACTINFO, UPDATE_USER,
+} from '../sources';
+import userRoleMapping from '../utils/constants';
+import ContactInfo from './ContactInfo';
+import PersonalData from './PersonalData';
+import CurriculumVitae from './CurriculumVitae';
+import './styles.css';
+import {
+  createSkills, createUserEducations, createUserExperiences, createUserSkills,
+} from './CreateMutationHelper';
+
+const MyProfilePage = () => {
+  const { userId, userRole } = useContext(AuthContext);
+
+  const { data } = useQuery(GET_USER, { variables: { id: userId } });
+  const [updateUser] = useMutation(UPDATE_USER);
+  const [updateContactInfo] = useMutation(UPDATE_CONTACTINFO);
+  const { data: countries } = useQuery(GET_COUNTRIES);
+  const [newSkills, setNewSkills] = useState([]);
+
+  const initialUser = {
+    username: data?.user.username,
+    password: data?.user.password,
+    firstName: data?.user.firstName,
+    lastName: data?.user.lastName,
+    role: data?.user.userRole.name,
+    email: data?.user.contactInfo.email,
+    phone: data?.user.contactInfo.phone,
+    city: data?.user.contactInfo.city,
+    country: data?.user.contactInfo.country.name,
+    website: data?.user.contactInfo.website,
+    avatarUrl: data?.user.contactInfo.avatarUrl,
+    about: data?.user.contactInfo.about,
+    userEducations: data?.user.userEducations,
+    userWorkExperiences: data?.user.userWorkExperiences,
+    userSkills: data?.user.userSkills,
+    createdAt: data?.user.createdAt,
+    updatedAt: data?.user.updatedAt,
+  };
+
+  const [updatedUser, setUpdatedUser] = useState(null);
+  const [personalInfoUpdated, setPersonalInfoUpdated] = useState(false);
+  const [contactInfoUpdated, setContactInfoUpdated] = useState(false);
+
+  const cvInitialState = {
+    education: false,
+    experience: false,
+    skills: false,
+  };
+
+  const [cvUpdated, setCvUpdated] = useState(cvInitialState);
+
+  const [updateUserEducation] = useMutation(
+    createUserEducations(
+      userId,
+      updatedUser?.userEducations?.filter((education) => education.id.toString().includes('new')),
+    ),
+  );
+  const [updateUserWorkExperience] = useMutation(
+    createUserExperiences(
+      userId,
+      updatedUser?.userWorkExperiences?.filter((experience) => experience.id.toString().includes('new')),
+    ),
+  );
+  const [createNewUserSkills] = useMutation(
+    createUserSkills(
+      userId,
+      newSkills,
+      updatedUser?.userSkills?.filter((skill) => skill.skill.id.toString().includes('new')),
```

Comment:
nitpick: you may rename filter arg to "entry" or something.

Reply:
updated